### ISSUE TITLE
Build failure with VS2015

### ISSUE THREAD

**Issue by nehaljwani** [mkleehammer/pyodbc#725]
### Environment

To diagnose, we usually need to know the following, including version numbers.  On Windows, be
sure to specify 32-bit Python or 64-bit:

- Python: 3.5+
- pyodbc: 4.0.30
- OS: Windows
- DB: N/A
- driver: N/A

### Issue

Observed behavior:

```

Failed to build pyodbc
Installing collected packages: pyodbc
  Created temporary directory: C:\Users\VssAdministrator\AppData\Local\Temp\pip-record-unehn5jr
    Running setup.py install for pyodbc: started
    Running command 'D:\bld\pyodbc_1583498944185\_h_env\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\VssAdministrator\\AppData\\Local\\Temp\\pip-req-build-hfnxvhr5\\setup.py'"'"'; __file__='"'"'C:\\Users\\VssAdministrator\\AppData\\Local\\Temp\\pip-req-build-hfnxvhr5\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\VssAdministrator\AppData\Local\Temp\pip-record-unehn5jr\install-record.txt' --single-version-externally-managed --compile --install-headers 'D:\bld\pyodbc_1583498944185\_h_env\Include\pyodbc'
    running install
    running build
    running build_ext
    building 'pyodbc' extension
    creating build
    creating build\temp.win-amd64-3.8
    creating build\temp.win-amd64-3.8\Release
    creating build\temp.win-amd64-3.8\Release\src
    C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\amd64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -DPYODBC_VERSION=4.0.30 -ID:\bld\pyodbc_1583498944185\_h_env\include -ID:\bld\pyodbc_1583498944185\_h_env\include "-IC:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\INCLUDE" "-IC:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\ATLMFC\INCLUDE" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\include\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\shared" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\winrt" -ID:\bld\pyodbc_1583498944185\_h_env\Library\include -ID:\bld\pyodbc_1583498944185\_h_env\Library\include /EHsc /Tpsrc\buffer.cpp /Fobuild\temp.win-amd64-3.8\Release\src\buffer.obj /Wall /wd4514 /wd4820 /wd4668 /wd4711 /wd4100 /wd4127 /wd4191 /d2FH4-
    buffer.cpp
    d:\bld\pyodbc_1583498944185\_h_env\include\cpython/abstract.h(77): warning C4365: 'return': conversion from 'std::size_t' to 'Py_ssize_t', signed/unsigned mismatch
    fatal error C1007: unrecognized flag '-FH4-' in 'p2'
    error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\BIN\\amd64\\cl.exe' failed with exit status 1
    Running setup.py install for pyodbc: finished with status 'error'
```

This is happening due to an assumption made by the fix for: https://github.com/mkleehammer/pyodbc/issues/663 , the assumption being that all python distributions (3.5 & above) have been built with VS2019+ , which is not valid, since the python distribution by conda-forge uses VS2015.

According to https://devblogs.microsoft.com/cppblog/making-cpp-exception-handling-smaller-x64/ , the `FH4` option/flag is a new addition which wasn't available in the previous versions.

**Comment by keitherskine** [comment]
Just FYI, I too found an issue with the fix for #663 with Python 3.5 and VS2015.  As part of my PR #697 I had to change line 171 of `setup.py` to include [only Python 3.6 and later](https://github.com/mkleehammer/pyodbc/pull/697/files#diff-2eeaed663bd0d25b7e608891384b7298).

**Comment by nehaljwani** [comment]
@keitherskine I don't think that's the correct fix. Conda Forge uses VS2015 for python 3.6 and 3.7

**Comment by keitherskine** [comment]
I take your point, @nehaljwani .  Microsoft hasn't made things easy by continually changing the way the VS2015/17/19 C++ compilers work.  This is partly why `setuptools.py` is so complicated.  What's more, it would be immensely helpful to know which version of MSVC `setup.py` is using when it runs the build.  However, as far as I can tell, `setuptools.py` does not export that info, it just figures it out at runtime.  (If anybody knows how to get that info, I'd be very interested.)  If we could know what version of MSVC is being used, then `setup.py` could tailor the compile flags to each version.

I know I'm not coming up with any answers here, @nehaljwani, just trying to explain the situation.

**Comment by hzhang-nib** [comment]
in tried to install an early pyodbc version  and it worked 

pip install -Iv pyodbc==4.0.27

**Comment by frjariello** [comment]
> in tried to install an early pyodbc version and it worked
> 
> pip install -Iv pyodbc==4.0.27

Thanks, I also solved it using this version with python 3.8.2

**Comment by keitherskine** [comment]
Closed due to inactivity. Feel free to re-open with current information if necessary.
